### ISSUE TITLE
Segmentation Fault on Windows 10

### ISSUE THREAD

**Issue by wheelerlaw** [genotrance/px#64]
I keep getting a `Segmentation Fault` when trying to hit google through my companies proxy server. 

Shell history:

```
$ https_proxy=http://localhost:3129 curl https://www.google.com
curl: (56) Recv failure: Connection was reset
$ https_proxy=http://localhost:3129 curl https://www.google.com
curl: (56) Recv failure: Connection was reset
```

px log:

```
$ px --proxy=upstream.proxy.hostname --port=3129 --noproxy=10.0.0.0/8,172.16.0.0/12,192.168.0.0/16 --de
bug
Serving at 127.0.0.1:3129 proc MainProcess
MainProcess: MainThread: 1536600059: print_banner: proxy:server = upstream.proxy.hostname
MainProcess: MainThread: 1536600059: print_banner: proxy:port = 3129
MainProcess: MainThread: 1536600059: print_banner: proxy:listen = 127.0.0.1
MainProcess: MainThread: 1536600059: print_banner: proxy:allow = *.*.*.*
MainProcess: MainThread: 1536600059: print_banner: proxy:gateway = 0
MainProcess: MainThread: 1536600059: print_banner: proxy:hostonly = 0
MainProcess: MainThread: 1536600059: print_banner: proxy:noproxy = 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
MainProcess: MainThread: 1536600059: print_banner: proxy:useragent =
MainProcess: MainThread: 1536600059: print_banner: settings:workers = 2
MainProcess: MainThread: 1536600059: print_banner: settings:threads = 5
MainProcess: MainThread: 1536600059: print_banner: settings:idle = 30
MainProcess: MainThread: 1536600059: print_banner: settings:socktimeout = 20.0
MainProcess: MainThread: 1536600059: print_banner: settings:proxyreload = 60
MainProcess: MainThread: 1536600059: print_banner: settings:foreground = 0
MainProcess: MainThread: 1536600059: print_banner: settings:log = 0
Serving at 127.0.0.1:3129 proc Process-1
Process-1: MainThread: 1536600060: print_banner: proxy:server = upstream.proxy.hostname
Process-1: MainThread: 1536600060: print_banner: proxy:port = 3129
Process-1: MainThread: 1536600060: print_banner: proxy:listen = 127.0.0.1
Process-1: MainThread: 1536600060: print_banner: proxy:allow = *.*.*.*
Process-1: MainThread: 1536600060: print_banner: proxy:gateway = 0
Process-1: MainThread: 1536600060: print_banner: proxy:hostonly = 0
Process-1: MainThread: 1536600060: print_banner: proxy:noproxy = 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
Process-1: MainThread: 1536600060: print_banner: proxy:useragent =
Process-1: MainThread: 1536600060: print_banner: settings:workers = 2
Process-1: MainThread: 1536600060: print_banner: settings:threads = 5
Process-1: MainThread: 1536600060: print_banner: settings:idle = 30
Process-1: MainThread: 1536600060: print_banner: settings:socktimeout = 20.0
Process-1: MainThread: 1536600060: print_banner: settings:proxyreload = 60
Process-1: MainThread: 1536600060: print_banner: settings:foreground = 0
Process-1: MainThread: 1536600060: print_banner: settings:log = 0
Process-1: MainThread: 1536600064: verify_request: Client address: 127.0.0.1
Process-1: Thread_0: 1536600064: do_CONNECT: Entering
Process-1: Thread_0: 1536600064: do_transaction: Entering
Process-1: Thread_0: 1536600064: get_destination: www.google.com:443
Process-1: Thread_0: 1536600064: get_destination: Couldn't resolve host
Process-1: Thread_0: 1536600064: do_socket_connect: New connection: ('upstream.proxy.hostname', 80)
Process-1: Thread_0: 1536600064: do_proxy_type: Searching proxy type
Process-1: Thread_0: 1536600064: do_socket: Entering
Process-1: Thread_0: 1536600064: do_socket: CONNECT www.google.com:443 HTTP/1.1
Process-1: Thread_0: 1536600064: do_socket: Sending Host: www.google.com:443
Process-1: Thread_0: 1536600064: do_socket: Sending User-Agent: curl/7.60.0
Process-1: Thread_0: 1536600064: do_socket: Sending Proxy-Connection: Keep-Alive
Process-1: Thread_0: 1536600064: do_socket: Reading response code
Process-1: Thread_0: 1536600064: do_socket: Response code: 407 True
Process-1: Thread_0: 1536600064: do_socket: Reading response headers
Process-1: Thread_0: 1536600064: do_socket: Received Proxy-Authenticate: sanitized (9)
Process-1: Thread_0: 1536600064: do_socket: Received Proxy-Authenticate: sanitized (4)
Process-1: Thread_0: 1536600064: do_socket: Received Proxy-Authenticate: sanitized (24)
Process-1: Thread_0: 1536600064: do_socket: Received Cache-Control: no-cache
Process-1: Thread_0: 1536600064: do_socket: Received Pragma: no-cache
Process-1: Thread_0: 1536600064: do_socket: Received Content-Type: text/html; charset=utf-8
Process-1: Thread_0: 1536600064: do_socket: Received Proxy-Connection: close
Process-1: Thread_0: 1536600064: do_socket: Received Connection: close
Process-1: Thread_0: 1536600064: do_socket: Received Content-Length: 1401
Process-1: Thread_0: 1536600064: do_proxy_type: Auth mechanisms: NEGOTIATE NTLM BASIC realm="Paychex_AD"
Process-1: Thread_0: 1536600064: do_proxy_type: Selected: ('upstream.proxy.hostname', 80): NTLM
Process-1: Thread_0: 1536600064: get_response_sspi: pywin32 SSPI
MainProcess: MainThread: 1536600066: verify_request: Client address: 127.0.0.1
MainProcess: Thread_0: 1536600066: do_CONNECT: Entering
MainProcess: Thread_0: 1536600066: do_transaction: Entering
MainProcess: Thread_0: 1536600066: get_destination: www.google.com:443
MainProcess: Thread_0: 1536600066: get_destination: Couldn't resolve host
MainProcess: Thread_0: 1536600066: do_socket_connect: New connection: ('upstream.proxy.hostname', 80)
MainProcess: Thread_0: 1536600066: do_proxy_type: Searching proxy type
MainProcess: Thread_0: 1536600066: do_socket: Entering
MainProcess: Thread_0: 1536600066: do_socket: CONNECT www.google.com:443 HTTP/1.1
MainProcess: Thread_0: 1536600066: do_socket: Sending Host: www.google.com:443
MainProcess: Thread_0: 1536600066: do_socket: Sending User-Agent: curl/7.60.0
MainProcess: Thread_0: 1536600066: do_socket: Sending Proxy-Connection: Keep-Alive
MainProcess: Thread_0: 1536600066: do_socket: Reading response code
MainProcess: Thread_0: 1536600066: do_socket: Response code: 407 True
MainProcess: Thread_0: 1536600066: do_socket: Reading response headers
MainProcess: Thread_0: 1536600066: do_socket: Received Proxy-Authenticate: sanitized (9)
MainProcess: Thread_0: 1536600066: do_socket: Received Proxy-Authenticate: sanitized (4)
MainProcess: Thread_0: 1536600066: do_socket: Received Proxy-Authenticate: sanitized (24)
MainProcess: Thread_0: 1536600066: do_socket: Received Cache-Control: no-cache
MainProcess: Thread_0: 1536600066: do_socket: Received Pragma: no-cache
MainProcess: Thread_0: 1536600066: do_socket: Received Content-Type: text/html; charset=utf-8
MainProcess: Thread_0: 1536600066: do_socket: Received Proxy-Connection: close
MainProcess: Thread_0: 1536600066: do_socket: Received Connection: close
MainProcess: Thread_0: 1536600066: do_socket: Received Content-Length: 1401
MainProcess: Thread_0: 1536600066: do_proxy_type: Auth mechanisms: NEGOTIATE NTLM BASIC realm="Paychex_AD"
MainProcess: Thread_0: 1536600066: do_proxy_type: Selected: ('upstream.proxy.hostname', 80): NTLM
MainProcess: Thread_0: 1536600066: get_response_sspi: pywin32 SSPI
Segmentation fault
```

Any ideas?

**Comment by wheelerlaw** [comment]
When I try to hit an internal site, I don't encounter any issues:

```
$ https_proxy=http://localhost:3129 curl https://internal.website.hostname
```

px log;

```
$ px --proxy=upstream.proxy.hostname --port=3129 --noproxy=10.0.0.0/8,172.16.0.0/12,192.168.0.0/16 --de
bug
Serving at 127.0.0.1:3129 proc MainProcess
MainProcess: MainThread: 1536600300: print_banner: proxy:server = upstream.proxy.hostname
MainProcess: MainThread: 1536600300: print_banner: proxy:port = 3129
MainProcess: MainThread: 1536600300: print_banner: proxy:listen = 127.0.0.1
MainProcess: MainThread: 1536600300: print_banner: proxy:allow = *.*.*.*
MainProcess: MainThread: 1536600300: print_banner: proxy:gateway = 0
MainProcess: MainThread: 1536600300: print_banner: proxy:hostonly = 0
MainProcess: MainThread: 1536600300: print_banner: proxy:noproxy = 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
MainProcess: MainThread: 1536600300: print_banner: proxy:useragent =
MainProcess: MainThread: 1536600300: print_banner: settings:workers = 2
MainProcess: MainThread: 1536600300: print_banner: settings:threads = 5
MainProcess: MainThread: 1536600300: print_banner: settings:idle = 30
MainProcess: MainThread: 1536600300: print_banner: settings:socktimeout = 20.0
MainProcess: MainThread: 1536600300: print_banner: settings:proxyreload = 60
MainProcess: MainThread: 1536600300: print_banner: settings:foreground = 0
MainProcess: MainThread: 1536600300: print_banner: settings:log = 0
Serving at 127.0.0.1:3129 proc Process-1
Process-1: MainThread: 1536600301: print_banner: proxy:server = upstream.proxy.hostname
Process-1: MainThread: 1536600301: print_banner: proxy:port = 3129
Process-1: MainThread: 1536600301: print_banner: proxy:listen = 127.0.0.1
Process-1: MainThread: 1536600301: print_banner: proxy:allow = *.*.*.*
Process-1: MainThread: 1536600301: print_banner: proxy:gateway = 0
Process-1: MainThread: 1536600301: print_banner: proxy:hostonly = 0
Process-1: MainThread: 1536600301: print_banner: proxy:noproxy = 10.0.0.0/8,172.16.0.0/12,192.168.0.0/16
Process-1: MainThread: 1536600301: print_banner: proxy:useragent =
Process-1: MainThread: 1536600301: print_banner: settings:workers = 2
Process-1: MainThread: 1536600301: print_banner: settings:threads = 5
Process-1: MainThread: 1536600301: print_banner: settings:idle = 30
Process-1: MainThread: 1536600301: print_banner: settings:socktimeout = 20.0
Process-1: MainThread: 1536600301: print_banner: settings:proxyreload = 60
Process-1: MainThread: 1536600301: print_banner: settings:foreground = 0
Process-1: MainThread: 1536600301: print_banner: settings:log = 0
Process-1: MainThread: 1536600314: verify_request: Client address: 127.0.0.1
Process-1: Thread_0: 1536600314: do_CONNECT: Entering
Process-1: Thread_0: 1536600314: do_transaction: Entering
Process-1: Thread_0: 1536600314: get_destination: internal.website.hostname:443
Process-1: Thread_0: 1536600314: get_destination: internal.website.hostname:443 => ('10.2.217.8', 443) + /
Process-1: Thread_0: 1536600314: get_destination: Direct connection from noproxy configuration
Process-1: Thread_0: 1536600314: do_transaction: Skipping NTLM proxying
Process-1: Thread_0: 1536600314: do_socket: Entering
Process-1: Thread_0: 1536600314: do_socket_connect: New connection: ('10.2.217.8', 443)
Process-1: Thread_0: 1536600314: do_CONNECT: Tunneling through proxy
Process-1: Thread_0: 1536600314: log_message: "CONNECT internal.website.hostname:443 HTTP/1.1" 200 -
Process-1: Thread_0: 1536600314: do_CONNECT: Connection closed by client
Process-1: Thread_0: 1536600314: do_CONNECT: Cleanup proxy connection
Process-1: Thread_0: 1536600314: do_CONNECT: 8140 bytes read, 8140 bytes written
Process-1: Thread_0: 1536600314: do_CONNECT: Done
```

**Comment by genotrance** [comment]
Can you please provide some info:
- Are you using Px release binary or running your own build?
  - If so what version of Python and pywin32 do you have?
- Windows OS version - build, arch

Seems like pywin32 is crashing on your system.

**Comment by wheelerlaw** [comment]
Python 3.7.0, px 0.4.0 (via pip install), Windows 10 build 14393.447

**Comment by genotrance** [comment]
Looks like multiple crashes with pywin32 on Python 3.7. I'm not sure which one is affecting Px or is it a new issue.

Can you please try with Python 3.6 if possible or use the binary distro?

https://github.com/mhammond/pywin32/issues/1221
https://github.com/mhammond/pywin32/issues/1233
https://github.com/mhammond/pywin32/issues/1234

**Comment by wheelerlaw** [comment]
Okay, sorry I forgot to circle around back to this. Changing to Python 3.6 fixed the issue.

**Comment by genotrance** [comment]
Thanks. I'll leave this open for now, we can close once there is a fix for 3.7.

**Comment by genotrance** [comment]
Looks like this should be fixed now.
